Homework7: Yura Buzinau

Eystei opened the pull request.

Eystei committed:
HW_7 feat: bulls_and_cows | buzzfuzz | like

additional updates:
- disable docstrings: pylint.yaml

Eystei committed:
HW_7 fix: fix type in likes.py

Eystei committed:
update pylint.yaml

Eystei committed:
Merge branch 'refs/heads/main' into features/HW_7

Stml89 commented on home_work/hw_7/like/likes.py:
```diff
@@ -0,0 +1,31 @@
+        if num_likes == 0:
+            return "no one likes this"
+
+        if num_likes == 1:
```
We do not want to execute all `if ` statements, because of time and resources, so it is better to use: `if-elif-else`.

Eystei replied:
fix

Eystei replied:
> We do not want to execute all `if ` statements, because of time and resources, so it is better to use: `if-elif-else`.

It won't enter each if because it will stop on the first return.

Stml89 commented on home_work/hw_7/like/likes.py:
```diff
@@ -0,0 +1,31 @@
+
+        names_list: list[str] = []
+        for name in names:
+            if isinstance(name, str):
```
What requirement does `if` statement cover?

Eystei replied:
removed unnecessary code. done

Stml89 commented on home_work/hw_7/bulls_and_cows_game/bulls_and_cows.py:
```diff
@@ -0,0 +1,36 @@
+                bulls += 1
+
+            if guess[i] in self.secret_number:
+                cows += 1
```
It will not count of cows correctly:
e.g.:
Secret: 7108 
Guess: 7180

Result: 2 bulls and 4 cows?

Eystei replied:
fix
![image](https://github.com/Eystei/pymar2024/assets/95315504/0666fc33-bf86-4fc8-bfcc-bc9c665b9234)

Stml89 commented on home_work/hw_7/bulls_and_cows_game/main.py:
```diff
@@ -0,0 +1,52 @@
+        if user_guess.lower() == "exit":
+            sys.exit()
+
+        if len(user_guess) != 4 or user_guess.isdigit() == 0:
```
I can enter `1122` as a guess

Eystei replied:
fix
![pycharm64_fIBQEB18zo](https://github.com/Eystei/pymar2024/assets/95315504/42399d12-4b24-46d0-ab5e-27164f8c7731)

Stml89 commented on home_work/hw_7/bulls_and_cows_game/bulls_and_cows.py:
```diff
@@ -0,0 +1,36 @@
+
+    def __init__(self):
+        """Initializes a new instance of the BullsAndCows class."""
+        self.secret_number = self.gen_four_digit_num()
```
If are not planning to generate random numbers more than ones, so let's move it here:
`self.secret_number = "".join(random.sample(string.digits, 4))`

Eystei replied:
moved
![image](https://github.com/Eystei/pymar2024/assets/95315504/9ee1ec88-b89b-454a-a3f0-da7c181fa45d)

Eystei committed:
HW_7 fix: remove module ascii_title.py

Eystei committed:
Merge remote-tracking branch 'origin/features/HW_7' into features/HW_7

Eystei committed:
Merge branch 'main' into features/HW_7

Eystei committed:
HW_7 fix: fixed bugs in the code found by Stml89

additional info:
- added check non-repeating 4-digit number in bulls_and_cows game
- fix bug in if statement in bulls_and_cows.py
- removed unnecessary code from likes.py
